lib/codecs/src/decoding/format/gelf.rs

```diff
@@ -0,0 +1,216 @@
+    // Builds a LogEvent from the parsed GelfMessage, adhering to GELF spec
+    fn insert_fields_from_gelf(&self, parsed: &GelfMessage) -> vector_core::Result<Event> {
+        let mut event = Event::from(parsed.short_message.to_string());
+        let log = event.as_mut_log();
+
+        // GELF spec defines the version as 1.1 which has not changed since 2013
+        if parsed.version != GELF_VERSION {
+            return Err(format!(
+                "{} does not match GELF spec version ({})",
+                VERSION, GELF_VERSION
+            )
+            .into());
+        }
+
+        log.insert(VERSION, parsed.version.to_string());
+        log.insert(HOST, parsed.host.to_string());
+
+        if let Some(full_message) = &parsed.full_message {
+            log.insert(FULL_MESSAGE, full_message.to_string());
+        }
+
+        if let Some(timestamp) = parsed.timestamp {
+            let naive = NaiveDateTime::from_timestamp(
+                f64::trunc(timestamp) as i64,
+                f64::fract(timestamp) as u32,
+            );
+            log.insert(
+                log_schema().timestamp_key(),
+                DateTime::<Utc>::from_utc(naive, Utc),
+            );
+        } else {
+            log.insert(log_schema().timestamp_key(), Utc::now());
+        }
+
+        if let Some(level) = parsed.level {
+            log.insert(LEVEL, level);
+        }
+        if let Some(facility) = &parsed.facility {
+            log.insert(FACILITY, facility.to_string());
+        }
+        if let Some(line) = &parsed.line {
+            log.insert(LINE, *line);
+        }
+        if let Some(file) = &parsed.file {
+            log.insert(FILE, file.to_string());
+        }
+
+        if let Some(add) = &parsed.additional_fields {
+            for (key, val) in add.iter() {
+                // per GELF spec, filter out _id
+                if key == "_id" {
+                    continue;
+                }
+                // per GELF spec, Additional field names must be characters dashes or dots
+
+                // drop fields names that don't match the GELF spec
+                if !self.regex.is_match(key) {
+                    continue;
+                }
+
+                // per GELF spec, values to additional fields can be strings or numbers
+                if val.is_string() {
+                    log.insert(path!(key.as_str()), val.as_str());
+                } else if val.is_u64() {
+                    log.insert(path!(key.as_str()), val.as_u64());
+                } else if val.is_i64() {
+                    log.insert(path!(key.as_str()), val.as_i64());
+                } else if val.is_f64() {
+                    match ordered_float::NotNan::new(val.as_f64().unwrap()) {
+                        Ok(float) => {
+                            log.insert(path!(key.as_str()), float);
+                        }
+                        Err(e) => return Err(e.to_string().into()),
+                    }
+                }
```

Comment:
Sorry for harping on this, but it does look like there is a `From<serde_json::Value>` implementation we can leverage for this: https://github.com/vectordotdev/vector/blob/91d88e8540bfe154038ca79265d28b31e6242c05/lib/value/src/value/serde.rs#L171-L191

Reply:
No need to apologize! After a closer look it looks like the issue I'm running into tis that that `From` impl is for a serde_json::Value, not a &serde_json::Value.
So in order to use that, I'd have to clone() it and that would be a wasteful copy :/
I'll see if I can find a way around that.